features.js

```diff
@@ -186,65 +186,52 @@ function getCodec(peerConnectionLog, kind, direction) {
     }
 }
 
-// extract a local/remote audio or video track.
-function extractTrack(peerConnectionLog, kind, direction) {
-    var trackId;
-    var reports = [];
-    var streamevent = 'onaddstream';
-    var trackevent = 'ontrack';
-    if (direction === 'send') {
-        streamevent = 'addStream';
-    }
-    // search for the (first) track of that kind.
-    for (var i = 0; i < peerConnectionLog.length; i++) {
-        var type = peerConnectionLog[i].type;
-        if (direction === 'send') {
-            if (type === 'addTrack') {
-                var kindAndTrack = peerConnectionLog[i].value.split(' ')[0].split(':');
-                if (kindAndTrack[0] === kind) {
-                    trackId = kindAndTrack[1];
-                    break;
-                }
-            }
-        } else {
-            if (type === streamevent || type === trackevent) {
-                if (type === trackevent) {
-                    var kindAndTrack = peerConnectionLog[i].value.split(' ')[0].split(':');
-                    if (kindAndTrack[0] === kind) {
-                        trackId = kindAndTrack[1];
-                        break;
-                    }
-                } else {
-                    var tracks = peerConnectionLog[i].value.split(' ', 2);
-                    tracks.shift();
-                    tracks = tracks[0].split(',');
-                    for (var j = 0; j < tracks.length; j++) {
-                        if (tracks[j].split(':')[0] === kind) {
-                            trackId = tracks[j].split(':')[1];
-                            break;
-                        }
+// extracts a Map with all local and remote audio/video tracks.
+function extractTracks(peerConnectionLog) {
+    const tracks = new Map();
+    for (let i = 0; i < peerConnectionLog.length; i++) {
+        const {type, value} = peerConnectionLog[i];
+        if (type === 'addStream') {
+            const [streamId, _] = value.split(' ');
+            _.split(',').forEach(id => {
+                const [kind, trackId] = id.split(':');
+                tracks.set(trackId, {kind, streamId, direction: 'send', stats: []});
+            });
+        } else if (type === 'addTrack') {
+            const [kind, trackId] = value.split(' ')[0].split(':');
+            const streamId = value.split(' ')[1].split(':')[1];
+            tracks.set(trackId, {kind, streamId, direction: 'send', stats: []});
+        } else if (type === 'ontrack') {
+            const [kind, trackId] = value.split(' ')[0].split(':');
```

Comment:
These 2 lines are duplicated, can we have a helper method?

function extractTrackInfoFromEvent(value) {
             const [kind, trackId] = value.split(' ')[0].split(':'); 
             const streamId = value.split(' ')[1].split(':')[1];
             return {kind, streamId, trackId};
}